File: CH5(greedy)/Activityـselection.py
n = int(input())

v = []

for i in range(n):
    l, r = map(int, input().split())
    v.append((l, r))

v.sort(key=lambda  x : x[1])

ans=0
last=-1

for l , r in v:
    if l>= last:
        last= r
        ans+=1

print(ans)

File: CH5(greedy)/Activityـselection.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){

    long long n;
    cin >> n;

    vector<pair<long long,long long>> v(n);

    for(int i=0;i<n;i++){
        cin >> v[i].first >> v[i].second;
    }

    sort(v.begin(), v.end(), [](auto& a, auto& b){
        return a.second < b.second;
    });


    long long ans = 0;
    long long last = -1;

    for(int i=0;i<n;i++){

        if(v[i].first >= last){
            ans++;
            last = v[i].second;
        }

    }

    cout << ans;

}